### ISSUE TITLE
Phantom marks left after delete

### ISSUE THREAD

**Issue by jpflouret** [kshenoy/vim-signature#72]
I have the following document:

      line 1
    a line 2
      line 3
    b line 4
      line 5

If I issue `:'a,'bdelete` I'm left with some phantom markers that I cannot get rid of. It looks like this:

      line 1
    a line 5

The `a` mark is not set in vim since I get `E20: mark not set` when I try to jump to it. `:SignatureRefresh` doesn't help. If I set the `a` mark somewhere else in the document the mark shows up correctly at the new place but does not get removed from the old place. It could look like this:

      line 1
    a line 5
      <more lines>
    a line 99

**Comment by kshenoy** [comment]
I hope the issue is resolved. If not, please feel free to reopen it.

**Comment by jamischarles** [comment]
I'm seeing this issue pretty frequently now. To repro here's what I do.

`ma` to clear the phantom mark. Close nvim. Reopen the same file, and the mark is back. Any suggestions?

In general are normal vim marks supposed to disappear when you close vim?

**Comment by kshenoy** [comment]
@jamischarles I'm not able to reproduce this error with vim. If I run `:'a,'bd`, there are no phantom marks left behind. 

I'll go home and try it out with nvim.

**Comment by jamischarles** [comment]
I did see this when I tried to clear a mark just right now `Signature: Unable to read/write .viminfo`

**Comment by kshenoy** [comment]
> I had to sudo to even see the contents of the file

That doesn't sound like an issue with the plugin itself but incorrect permissions on the .viminfo file itself. Vim uses .viminfo to save information such as command line history, search history, marks etc. Signature just reads and writes to the file to ensure that global marks are correctly removed.

Could you run `ls -l ~/.viminfo` and post the output?

**Comment by jamischarles** [comment]
```
→ ls -l ~/.viminfo
-rw-------  1 root  110187567  1023 Sep 12 10:46 /Users/jacharles/.viminfo
```

I'm going to wipe it and recreate it with touch and see what effect that has. What are the expected permissions for `.viminfo`?

**Comment by jamischarles** [comment]
This might explain it. Looks like neovim doesn't use .viminfo anymore: https://www.reddit.com/r/neovim/comments/5con51/where_is_neovims_viminfo_located/

**Comment by jamischarles** [comment]
I tried wiping my .viminfo, then symlinking my main.shada file to the .viminfo file, but the issue still persists even with that.

**Comment by jamischarles** [comment]
Related I think https://github.com/neovim/neovim/issues/4288

**Comment by jamischarles** [comment]
tl;dr: Sounds like the hacky workaround is for meto call `:wshada!` before I exit the file, or after I change marks to persist those mark deletions.

**Comment by jamischarles** [comment]
Those threads imply that this behavior was present with viminfo files as well. Wonder if we can apply your same workaround here and call `:wshada!` in the same place?

**Comment by jamischarles** [comment]
I see in this file https://github.com/kshenoy/vim-signature/blob/master/autoload/signature/mark.vim#L324 that you're manually adjusting the `.viminfo` file. I'd be happy to issue a PR to fix this for neovim. Here's what I'm thinking:

1) Check if you're in neovim.
2) On global / local mark remove, run `:wsh` command (or we could manually remove entry from shada file (neovim equivalent of viminfo)

Thoughts?

```
:wsh[ada][!] [file]	Write to ShaDa file [file] (default: see above).
			The information in the file is first read in to make
			a merge between old and new info.  When [!] is used,
			the old information is not read first, only the
			internal info is written (also disables safety checks 
			described in |shada-error-handling|).  If 'shada' is 
			empty, marks for up to 100 files will be written.
			When you get error "E929: All .tmp.X files exist, 
			cannot write ShaDa file!" check that no old temp files 
			were left behind (e.g. 
			~/.local/share/nvim/shada/main.shada.tmp*).

			Note: Executing :wshada will reset all |'quote| marks.
```

**Comment by kshenoy** [comment]
> Check if you're in neovim.

Ah nevermind, that's the problem. For some reason I assumed you were using both vim and neovim (even though you didn't mention anything about using vim) and the mention of .viminfo caused me to lead you on a wild goose chase. My apologies for that and thanks for the debug.

I updated that function `s:ForceGlobalMarkRemoval` to work with neovim. However, note that the methodology to remove the global mark is different than earlier and may have much bigger side-effects so it's been disabled by default. Set `g:ForceRemoveGlobalMark` to 1 if you want to use this.
